### PR TITLE
Prerequisites endpoint update

### PR DESCRIPTION
**Pull request by giraffecrackers**
Updating prerequisite endpoint to use standard transaction object shape
Main changes:
• No more erc20_approval type
• smart_contract_write is used for all write interactions with smart contract
• Updated `Issue a Virtual card` guide

This is only a part of the change. All parts should be merged together
Here are links to other PRs:
https://github.com/immersve/borax/pull/213
https://github.com/immersve/amethyst/pull/766

Ticket Link: https://www.notion.so/immersve/Approval-prerequisite-includes-ERC20-address-f3b641b898c64c5fae8557c34228e957?pvs=4

## Screenshots
image
image
image

### CODE REVIEW

File: docs/guides/non-custodial wallets/issue-a-virtual-card.md [hunk 1]
```diff
@@ -24,17 +24,72 @@ A user will more often than not be quoted a price for a purchase by a merchant i
...
+```ts
+import { JsonFragment } from '@ethersproject/abi';
+/**
+ * Builds array of params to be used in calls to smart contracts
+ * @param abi ABI for smart contact function
+ * @param method function name to be called
+ * @param params object of properties matching to the input names specified in ABI
+ * @returns array of params in correct order
+ */
+export function toSmartContractParams(
```
Comment (ncjones) [inline docs/guides/non-custodial wallets/issue-a-virtual-card.md:54]: I think this function is not required; just return `Object.values(params)`

Comment (giraffecrackers) [reply docs/guides/non-custodial wallets/issue-a-virtual-card.md:54]: This function gives confidence in order of the params

Comment (giraffecrackers) [reply docs/guides/non-custodial wallets/issue-a-virtual-card.md:54]: found this https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order and decided to make sure order is correct

**Comment by catluin** [comment]
Do we need to communicate this change to BoF?

**Comment by giraffecrackers** [comment]
> Do we need to communicate this change to BoF?

I think yes. it will not hurt anyway

File: docs/guides/non-custodial wallets/issue-a-virtual-card.md (hunk 1 above)
Comment (ncjones) [reply docs/guides/non-custodial wallets/issue-a-virtual-card.md:53]: We need to make our APIs easy to use. The order should be guaranteed on the server not on the client.

File: openapi/models/smart-contract-write-params.yaml [hunk 2]
```diff
@@ -4,11 +4,45 @@ properties:
...
+              description: The type of the parameter
+              type: string
+              example: uint256
   method:
     type: string
     description: Smart contract method
   params:
     type: object
-    description: Smart contract call params
+    description: Smart contract call params. Keys are matching input names defined in ABI
```
Comment (ncjones) [inline openapi/models/smart-contract-write-params.yaml:48]: Keys are matching names and order defined in ABI

Comment (giraffecrackers) [reply openapi/models/smart-contract-write-params.yaml]: ty. updated.

File: docs/guides/non-custodial wallets/issue-a-virtual-card.md (hunk 1 above)
Comment (giraffecrackers) [reply docs/guides/non-custodial wallets/issue-a-virtual-card.md]: removed this block in favour of 
```ts
Object.values(params)
```

**Review by ncjones — APPROVED** [review]

**Pull request merged**